dialect/sql/schema/atlas.go

```diff
@@ -418,21 +662,55 @@ func (m *Migrate) atDiff(ctx context.Context, conn dialect.ExecQuerier, name str
 	if err != nil {
 		return nil, err
 	}
-	tt, err := m.aTables(ctx, m, conn, tables)
+	var types []string
+	if a.universalID {
+		// Fetch pre-existing type allocations.
+		exists, err := a.sqlDialect.tableExist(ctx, conn, TypeTable)
+		if err != nil {
+			return nil, err
+		}
+		if exists {
+			rows := &entsql.Rows{}
+			query, args := entsql.Dialect(a.dialect).
+				Select("type").From(entsql.Table(TypeTable)).OrderBy(entsql.Asc("id")).Query()
+			if err := conn.Query(ctx, query, args, rows); err != nil {
+				return nil, fmt.Errorf("query types table: %w", err)
+			}
+			defer rows.Close()
+			a.types = nil
+			if err := entsql.ScanSlice(rows, &a.types); err != nil {
+				return nil, err
+			}
+		}
+		types = a.types
+	}
+	desired, err := a.StateReader(tables...).ReadState(ctx)
```

Comment:
You already have the schema name at this stage if you want to append it to the state-reader

Reply:
Yes, I prefer to have the `StateReader` not care about this since Ent itself doesn't know about its schema.